src/Nethermind/Nethermind.Serialization.Rlp/TxDecoder.cs

```diff
@@ -25,7 +25,7 @@ public TxDecoder(bool lazyHash) : base(lazyHash)
 
         public int GetLength(Transaction tx)
         {
-            return GetLength(tx, RlpBehaviors.None);
+            return GetLength(tx, RlpBehaviors.InMempoolForm);
```

Comment:
The method name does not seem safe. Could we reflect somehow that it's used by mempool gossip only or like that. Otherwise it's too easy to use this method when execution-payload form is expected. For example `Roundtrip_ExecutionPayloadForm_for_shard_blobs` seems to work just by accident.

Reply:
Good point. I renamed it to `GetNetworkTxLength` to avoid confusion on `TxDecoder` level. It's still called as `tx.GetLength()` from `TxPool` endpoint as `GetLength()` is a part of `TransactionExtensions` for `TxPool`, but I think it doesn't matter as it can be called only from `TxPool` project. Shared code is now renamed. I'm not 100% happy about this naming, if you have an idea for a better name feel free to propose :)

Reply:
Seems OK. But there is another idea. What if we rename `ITransactionSizeCalculator` to `IMempoolTransactionSizeCalculator`. And implement `GetLength` explicitly as a private method, so `GetLength` name may stay and this method will not contaminate TxDecoder API.